### DISCUSSION TITLE
`iterate_properties` method of a Mapper does not include `backref` properties

### DISCUSSION THREAD

**Discussion by michaelbukachi** [sqlalchemy/sqlalchemy#9489] [Usage Questions] [answered]
The `iterate_properties` of a mapper no longer includes relationship properties from `backrefs`. Take a look at the following snippet:
```python
class User(Base):
    __tablename__ = 'user'
    id = Column(Integer, primary_key=True)
    posts = relationship('Address', backref='user')


class Address(Base):
    __tablename__ = 'address'
    id = Column(Integer, primary_key=True)
    user_id = Column(Integer, ForeignKey('user.id'))

if __name__ == '__main__':
    print([c.key for c in Address.__mapper__.iterate_properties
                if isinstance(c, RelationshipProperty)])

```
SQLAlchemy < 2.0:
```
# prints out ['user']
```

SQLAlchemy >= 2.0:
```
# prints out []
```
Is there a way to access relationship properties using the new API?

**Comment by CaselIT** [comment]
hi,

Can you check if you get the same behaviour after calling configure_mappers before looping over the properties?

**Reply by michaelbukachi** [reply]
Interesting. Calling `configure_mappers` produces the same behaviour as `< 2.0`. Does this mean that pre-2.0, `configure_mappers` was called implicitly?

**Comment by zzzeek** [comment, marked as answer]
this was an intentional change as part of #9220 and is in the changelog at https://docs.sqlalchemy.org/en/20/changelog/changelog_20.html#change-45fe5664c88893d899c991ca3675142c .  this is not really a "public" method and should be changed to be underscored.  use [mapper.attrs](https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.Mapper.attrs) and related collections for public-facing inspection of mapped properties.
